botcore/_bot.py

```diff
@@ -209,15 +213,18 @@ async def setup_hook(self) -> None:
         """
         loop = asyncio.get_running_loop()
 
+        self._guild_available = asyncio.Event()
+
         self._resolver = aiohttp.AsyncResolver()
         self._connector = aiohttp.TCPConnector(
             resolver=self._resolver,
             family=socket.AF_INET,
         )
         self.http.connector = self._connector
 
-        self._connect_statsd(self.statsd_url, loop)
+        # Create dummy stats client first, in case `statsd_url` is unreachable within `_connect_statsd()`
         self.stats = AsyncStatsClient(loop, "127.0.0.1")
+        self._connect_statsd(self.statsd_url, loop)
```

Comment:
Why not create the dummy inside `_connect_statsd` if maximum attempts were reached? Also is this dummy just going to void requests? it won't error or anything?

Reply:
Since `_connect_statsd` calls itself on failure up to 8 times, so that would cause the dummy to be created up to 8 times.

We need to create the dummy before erroring since the callback is done with a loop.call_later, so in theory some cogs could already be loaded and trying to use statsd.

The dummy does just void and stats, no errors.